tests/unit/test_converters.py

```diff
@@ -0,0 +1,90 @@
+"""
+Tests the converters
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import mock
+import os
+import unittest
+
+import pysam
+
+import ga4gh.protocol as protocol
+import ga4gh.converters as converters
+import tests.utils as utils
+
+
+class TestSamConverter(unittest.TestCase):
+    """
+    Base class for SAM converter tests.
+    Provides common  methods.
+    """
+    readsFilePath = 'tests/unit/reads.dat'
+
+    def _getReads(self):
+        readsFile = file(self.readsFilePath)
+        lines = readsFile.readlines()
+        reads = []
+        for line in lines:
+            read = protocol.GAReadAlignment.fromJsonString(line)
+            reads.append(read)
+        return reads
+
+    def getHttpClient(self):
+        httpClient = utils.makeHttpClient()
+        httpClient.searchReads = self.getSearchReadsResponse
+        return httpClient
+
+    def getSearchReadsRequest(self):
+        request = protocol.GASearchReadsRequest()
+        return request
+
+    def getSearchReadsResponse(self, request):
+        return self._getReads()
+
+
+class TestSamConverterLogic(TestSamConverter):
+    """
+    Test the SamConverter logic
+    """
+    def testSamConverter(self):
+        mockPysam = mock.Mock()
+        with mock.patch('pysam.AlignmentFile', mockPysam):
+            httpClient = self.getHttpClient()
+            searchReadsRequest = self.getSearchReadsRequest()
+            outputFile = None
+            binaryOutput = False
+            samConverter = converters.SamConverter(
+                httpClient, searchReadsRequest, outputFile, binaryOutput)
+            samConverter.convert()
+
+
+class TestSamConverterRoundTrip(TestSamConverter):
+    """
+    Write a sam file and see if pysam can read it
+    """
+    def _testRoundTrip(self, binaryOutput, filePath):
+        # write SAM file
+        httpClient = self.getHttpClient()
+        searchReadsRequest = self.getSearchReadsRequest()
+        binaryOutput = False
+        samConverter = converters.SamConverter(
+            httpClient, searchReadsRequest, filePath, binaryOutput)
+        samConverter.convert()
+
+        # read SAM file
+        samfile = pysam.AlignmentFile(filePath, "r")
+        reads = [_ for _ in samfile.fetch()]
+        self.assertEqual(reads[0].query_name, "SRR622461.77861202")
+        samfile.close()
+
+        # cleanup
+        os.remove(filePath)
+
+    def testPlainText(self):
+        self._testRoundTrip(False, 'sam.sam')
```

Comment:
We should use a tempfile here, as we shouldn't ever write temporary files to CWD. Something like
```
with tempfile.NamedTemporaryFile() as fileHandle:
    self._testRoundTrip(False, fileHandle.name)
```